front/src/modules/trainschedule/components/ImportTrainSchedule/generateTrainSchedulesPayload.ts

```diff
@@ -46,3 +56,66 @@ export default function generateTrainSchedulesPayload(
   });
   return Object.values(trainSchedulesByPathID);
 }
+
+export function generateV2TrainSchedulesPayload(
+  trains: TrainSchedule[],
+  timetableID: number
+): TrainScheduleForm[] {
+  function formatToISO8601(dateTimeStr: string) {
+    const date = new Date(dateTimeStr);
+
+    // Get the timezone offset in minutes and convert it to hh:mm format
+    const offset = date.getTimezoneOffset();
+    const offsetHours = Math.abs(Math.floor(offset / 60));
+    const offsetMinutes = Math.abs(offset % 60);
+    const timezoneFormatted = `${offset > 0 ? '-' : '+'}${offsetHours.toString().padStart(2, '0')}:${offsetMinutes.toString().padStart(2, '0')}`;
+
+    // Format the date to ISO string without 'Z' (UTC) and add the timezone
+    const isoDateTime = `${date.toISOString().replace('Z', '')}${timezoneFormatted}`;
+
+    return isoDateTime;
+  }
+
+  function calculateTimeDifferenceInSeconds(time1: string, time2: string) {
+    const date1 = new Date(time1);
+    const date2 = new Date(time2);
+    return (date2.getTime() - date1.getTime()) / 1000;
+  }
+
+  function formatDurationAsISO8601(seconds: number) {
+    return `PT${Math.abs(seconds)}S`;
+  }
+
+  return trains.map((train) => {
+    const { path, schedule } = train.steps.reduce(
+      (acc, step, index) => {
+        const stepId = nextId();
+
+        acc.path.push({ id: stepId, uic: Number(step.uic) });
+        if (acc.schedule && index !== 0) {
+          const timeDifferenceInSeconds = calculateTimeDifferenceInSeconds(
+            train.departureTime,
+            step.arrivalTime
+          );
+          const isoDuration = formatDurationAsISO8601(timeDifferenceInSeconds);
+          acc.schedule.push({
+            at: stepId,
+            arrival: isoDuration,
+            stop_for: `PT${step.duration}S`,
```

Comment:
You should specify the stop_for only if the step.duration is not undefined :) You can do it like this:
```suggestion
            ...(step.duration ? { stop_for: `PT${step.duration}S` } : {})
```